src/test/java/com/amazonaws/encryptionsdk/internal/FrameEncryptionHandlerTest.java

```diff
@@ -117,4 +132,87 @@ private void assertHeaderNonce(byte[] expectedNonce, byte[] buf) {
   private void generateTestBlock(byte[] buf) {
     frameEncryptionHandler_.processBytes(new byte[frameSize_], 0, frameSize_, buf, 0);
   }
+
+  /**
+   * This isn't a unit test, but it reproduces a bug in the FrameEncryptionHandler where the stream
+   * would be truncated when the offset is >0
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testStreamTruncation() throws Exception {
```

Comment:
The test could use some more comments to explain how you are verifying what is going on.